services/provider/server/server.go

```diff
@@ -557,12 +563,34 @@ func validateTicket(ticket string, pubKey *rsa.PublicKey) error {
 // FulfillStorageClassClaim RPC call to create the StorageclassClaim CR on
 // provider cluster.
 func (s *OCSProviderServer) FulfillStorageClassClaim(ctx context.Context, req *pb.FulfillStorageClassClaimRequest) (*pb.FulfillStorageClassClaimResponse, error) {
-	return &pb.FulfillStorageClassClaimResponse{}, nil
+	// Get storage consumer resource using UUID
+	consumerObj, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+
+	klog.Infof("Found storageConsumer UUID %q with Name %q", req.StorageConsumerUUID, consumerObj.Name)
+
+	claimID, err := s.storageClassClaimManager.Create(ctx, req.StorageConsumerUUID, req.StorageClassClaimName, req.StorageType.String(), req.EncryptionMethod)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return nil, status.Errorf(codes.Internal, "failed to FulfillStorageClassClaim for %q. %v", req.StorageConsumerUUID, err)
+		}
+	}
+
+	return &pb.FulfillStorageClassClaimResponse{
+		StorageClassClaimID: claimID,
+	}, nil
```

Comment:
This is useful for tracking purposes, even though for today it's not much useful. we can store the claimUUID on the claim Object on the consumer side for identity purposes. If we change the implementation  (breaking changes) this might help RPC calls for resource identification.

Reply:
Why? we already know how to get from a claim name and consumer UID to a remote claim id

Reply:
that is on the provider side not on the consumer side. Dont we need any tracking on the consumer side?

Reply:
Removed claimID from the response forgot to update here.

Reply:
I think exposing provider implementation details on consumer is problamatic